newrelic/metric.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package newrelic
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/google/ts-bridge/record"
+
+	"github.com/golang/protobuf/ptypes"
+	"google.golang.org/appengine/log"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// MetricConfig defines configuration file parameters for a specific metric imported from New Relic.
+type MetricConfig struct {
+	APIKey        string `yaml:"api_key" validate:"nonzero"`
+	ApplicationId string `yaml:"application_id" validate:"nonzero"`
+	MetricName    string `validate:"nonzero" yaml:"metric_name"`
+	MetricValue   string `validate:"nonzero" yaml:"metric_value"`
+	Cumulative    bool
```

Comment:
I am not seeing any support for cumulative metrics. Do you need this field?

Reply:
nope.